src/main/java/org/folio/rest/impl/EholdingsPackagesImpl.java

```diff
@@ -218,29 +225,73 @@ public void deleteEholdingsPackagesByPackageId(String packageId, Map<String, Str
+                                                       int page, int count, Map<String, String> okapiHeaders,
+                                                       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
-    FilterQuery fq = FilterQuery.builder()
-      .selected(RestConstants.FILTER_SELECTED_MAPPING.get(filterSelected))
-      .type(filterType)
-      .name(filterName).isxn(filterIsxn).subject(filterSubject)
-      .publisher(filterPublisher).build();
+    RMAPITemplate template = templateFactory.createTemplate(okapiHeaders, asyncResultHandler);
 
-    titleParametersValidator.validate(fq, sort, true);
+    if (isTagOnlySearch(filterTags, filterSelected, filterType, filterName, filterIsxn, filterSubject, filterPublisher)) {
+      List<String> tags = parseTags(filterTags);
+      template.requestAction(context -> getTitlesByPackageIdAndTags(packageId, tags, page, count, context));
+    } else {
 
-    Sort nameSort = Sort.valueOf(sort.toUpperCase());
+      PackageId parsedPackageId = idParser.parsePackageId(packageId);
 
-    templateFactory.createTemplate(okapiHeaders, asyncResultHandler)
-      .requestAction(context ->
-        context.getTitlesService().retrieveTitles(parsedPackageId.getProviderIdPart(), parsedPackageId.getPackageIdPart(), fq, nameSort, page, count)
-      )
+      FilterQuery fq = FilterQuery.builder()
+        .selected(RestConstants.FILTER_SELECTED_MAPPING.get(filterSelected))
+        .type(filterType)
+        .name(filterName)
+        .isxn(filterIsxn)
+        .subject(filterSubject)
+        .publisher(filterPublisher)
+        .build();
+
+      titleParametersValidator.validate(fq, sort, true);
+
+      Sort nameSort = Sort.valueOf(sort.toUpperCase());
+
+      template
+        .requestAction(context -> context.getTitlesService()
+          .retrieveTitles(parsedPackageId.getProviderIdPart(), parsedPackageId.getPackageIdPart(), fq, nameSort, page, count)
+        );
+    }
+    template
       .addErrorMapper(ResourceNotFoundException.class, exception ->
         GetEholdingsPackagesResourcesByPackageIdResponse.respond404WithApplicationVndApiJson(
           ErrorUtil.createError(PACKAGE_NOT_FOUND_MESSAGE)))
       .executeWithResult(ResourceCollection.class);
   }
 
+  private CompletableFuture<ResourceCollectionResult> getTitlesByPackageIdAndTags(String packageId, List<String> tags, int page, int count, RMAPITemplateContext context) {
+    MutableObject<Integer> totalResults = new MutableObject<>();
+    MutableObject<List<DbResource>> mutableDbTitles = new MutableObject<>();
+    String tenant = context.getOkapiData().getTenant();
+
+    return tagRepository
+      .countRecordsByTagsAndPrefix(tags, packageId + "-", tenant, RecordType.RESOURCE)
+      .thenCompose(resourceCount -> {
+        totalResults.setValue(resourceCount);
+        return resourceRepository.getResourcesByTagNameAndPackageId(tags, packageId, page, count, tenant);
```

Comment:
We can load holdings data for each resource from local db,
then we can only send requests to RM API for those resources that are missing from local db.

Reply:
@andrii-paias thanks for the note, added